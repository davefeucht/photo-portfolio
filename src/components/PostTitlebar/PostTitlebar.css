.post-titlebar {
  background-color: #000000;
  font-size: 25px;
  color: #FFFFFF;
  grid-area: header;
  display: grid;
  grid-template-columns: 95% 5%;
  grid-template-rows: 100%;
}

.post-titlebar > .title {
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  margin: auto auto auto 5px;
}