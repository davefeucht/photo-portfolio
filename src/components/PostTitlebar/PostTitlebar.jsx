/** **************
* PostTitlebar component displays the titlebar for a Post
*************** */

import './PostTitlebar.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React from 'react';

import CloseButton from '../CloseButton/CloseButton.jsx';

function PostTitlebar({ postTitle }) {
    return (
        <div className="post-titlebar">
            <div className="title">{postTitle}</div>
            <CloseButton />
        </div>
    );
}

PostTitlebar.displayName = 'PostTitlebar';

PostTitlebar.propTypes = {
    postTitle: PropTypes.string.isRequired
};

export default observer(PostTitlebar);
