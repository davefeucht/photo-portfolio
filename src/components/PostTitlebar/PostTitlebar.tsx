/** **************
* PostTitlebar component displays the titlebar for a Post
*************** */

import './PostTitlebar.css';

import { observer } from 'mobx-react';
import * as React from 'react';

import CloseButton from '../CloseButton/CloseButton';

interface PostTitlebarProps {
    postTitle: string
}

const PostTitlebar: React.FC<PostTitlebarProps> = ({ postTitle = null }) => {
    return (
        <div className="post-titlebar">
            <div className="title">{postTitle}</div>
            <CloseButton />
        </div>
    );
};

PostTitlebar.displayName = 'PostTitlebar';

export default observer(PostTitlebar);
