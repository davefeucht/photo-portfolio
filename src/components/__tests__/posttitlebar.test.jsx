import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import PostTitlebar from '../PostTitlebar/PostTitlebar';

const postTitle = 'Some Post';

test('PostTitlebar displays', () => {
    const component = renderer.create(
        <MemoryRouter>
            <PostTitlebar postTitle={postTitle} />
        </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
