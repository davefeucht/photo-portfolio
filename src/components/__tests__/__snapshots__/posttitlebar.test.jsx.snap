// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTitlebar displays 1`] = `
<div
  className="post-titlebar"
>
  <div
    className="title"
  >
    Some Post
  </div>
  <div
    className="close-button"
  >
    <a
      href="/category/undefined"
      onClick={[Function]}
    >
      <div
        className="close-button-content"
      >
        <img
          height="30px"
          src="./assets/images/close-button.svg"
          width="30px"
        />
      </div>
    </a>
  </div>
</div>
`;
